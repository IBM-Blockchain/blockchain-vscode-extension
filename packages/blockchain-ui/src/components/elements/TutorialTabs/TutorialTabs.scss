.tutorial-tabs-background {
    width: 100%;
    height: 48px;
    position: absolute;
    left: 0px;
    z-index: -1;
    background-color: white;
}



.bx--tab-content {
    width: 100%;
    padding-left: 0px;
    background-color: transparent;
}
