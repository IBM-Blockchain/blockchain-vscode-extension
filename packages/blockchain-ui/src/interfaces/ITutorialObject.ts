interface ITutorialObject {
    title: string;
    series: string;
    length: string;
    firstInSeries?: boolean;
    file: string;
    objectives: string[];
}

export default ITutorialObject;
