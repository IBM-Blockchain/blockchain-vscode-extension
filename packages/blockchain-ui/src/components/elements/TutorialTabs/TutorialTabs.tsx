import React, { Component } from 'react';
import { Tabs, Tab } from 'carbon-components-react';
import './TutorialTabs.scss';
import TutorialTile from '../TutorialTile/TutorialTile';

interface IProps {
    tutorialData: Array<{seriesName: string, seriesTutorials: any[]}>;
}

class TutorialTabs extends Component<IProps> {

    createTabs(): Array<JSX.Element> {
        const tabArray: JSX.Element[] = [];
        this.props.tutorialData.map((tutorialSeries: {seriesName: string, seriesTutorials: any[]}, index: number) => {
            const tabLabel: string = `${tutorialSeries.seriesName} (${tutorialSeries.seriesTutorials.length})`;
            tabArray.push(
                // @ts-ignore
                <Tab
                    href='#'
                    tabIndex={index}
                    label={tabLabel}
                >
                    {this.populateTabs(tutorialSeries.seriesTutorials)}
                </Tab>
            );
        });
        return tabArray;
    }

    populateTabs(seriesTutorials: any[]): Array<JSX.Element> {
        const tutorialNameArray: JSX.Element[] = [];
        for (const tutorial of seriesTutorials) {
            tutorialNameArray.push(
                <TutorialTile tutorialObject={tutorial}/>
            );
        }
        return tutorialNameArray;
    }

    render(): JSX.Element {
        let tabs: JSX.Element[] | JSX.Element;

        if (this.props.tutorialData.length > 0) {
            tabs = (
                // @ts-ignore
                <Tabs triggerHref='#' type='container'>
                    {this.createTabs()}
                </Tabs>
            );
        } else {
            tabs = (<></>);
        }

        return (
            <>
                <div className='tutorial-tabs-background'></div>
                {tabs}
            </>
        );
    }
}

export default TutorialTabs;
