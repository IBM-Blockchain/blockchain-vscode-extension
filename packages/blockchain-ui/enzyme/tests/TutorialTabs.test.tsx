import React from 'react';
import renderer from 'react-test-renderer';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import TutorialTabs from '../../src/components/elements/TutorialTabs/TutorialTabs';
import ITutorialObject from '../../src/interfaces/ITutorialObject';

chai.should();
chai.use(sinonChai);

describe('TutorialTabs component', () => {

    const tutorialData: Array<{name: string, tutorials: ITutorialObject[]}> = [
        {
            name: 'Basic tutorials',
            tutorials: [
                {
                    title: 'a1',
                    series: 'Basic tutorials',
                    length: '4 weeks',
                    objectives: [
                    'objective 1',
                    'objective 2',
                    'objective 3'
                    ],
                    file: 'some/file/path'
                }
            ]
        },
        {
            name: 'Other tutorials',
            tutorials: [
                {
                    title: 'something really interesting',
                    series: 'Other tutorials',
                    length: '10 minutes',
                    objectives: [
                        'objective 1',
                        'objective 2',
                        'objective 3'
                    ],
                    file: 'another/file/path'
                }
            ]
        }
    ];

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<TutorialTabs tutorialData={tutorialData}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the expected snapshot when no tutorial data is received', () => {
        const component: any = renderer
            .create(<TutorialTabs tutorialData={[]}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
