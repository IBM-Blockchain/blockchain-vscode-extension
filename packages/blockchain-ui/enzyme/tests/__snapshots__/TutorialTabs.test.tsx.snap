// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialTabs component should render the expected snapshot 1`] = `
Array [
  <div
    className="tutorial-tabs-background"
  />,
  <div
    className="bx--tabs bx--tabs--container"
    role="navigation"
    selected={0}
  >
    <div
      aria-label="listbox"
      className="bx--tabs-trigger"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <a
        className="bx--tabs-trigger-text"
        href="#"
        onClick={[Function]}
        tabIndex={-1}
      >
        Basic tutorials (1)
      </a>
      <svg
        aria-hidden={true}
        focusable="false"
        height={16}
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="8,11 3,6 3.7,5.3 8,9.6 12.3,5.3 13,6"
        />
        <title>
          show menu options
        </title>
      </svg>
    </div>
    <ul
      className="bx--tabs__nav bx--tabs__nav--hidden"
      role="tablist"
    >
      <li
        aria-controls="undefined__panel"
        aria-selected={true}
        className="bx--tabs__nav-item bx--tabs__nav-item--selected"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <a
          className="bx--tabs__nav-link"
          href="#"
          tabIndex={0}
        >
          Basic tutorials (1)
        </a>
      </li>
      <li
        aria-controls="undefined__panel"
        aria-selected={false}
        className="bx--tabs__nav-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <a
          className="bx--tabs__nav-link"
          href="#"
          tabIndex={-1}
        >
          Other tutorials (1)
        </a>
      </li>
    </ul>
  </div>,
  <div
    aria-hidden={false}
    className="bx--tab-content"
    hidden={false}
    role="tabpanel"
    selected={true}
  >
    <div
      className="tab-container"
    >
      <div
        className="tutorial-title"
      >
        a1
      </div>
      <div
        className="time-container"
      >
        <svg
          aria-hidden={true}
          className="icon"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="9"
            width="2"
            x="15"
            y="11"
          />
          <rect
            height="2"
            width="6"
            x="13"
            y="2"
          />
          <path
            d="M28,9,26.58,7.59,24.33,9.84a10.94,10.94,0,1,0,1.18,1.65ZM16,26a9,9,0,1,1,9-9A9,9,0,0,1,16,26Z"
          />
        </svg>
        <span
          className="time-text"
        >
          4 weeks
        </span>
      </div>
      <div
        className="tutorial-objectives"
      >
        <p
          className="objective"
        >
          objective 1
        </p>
        <p
          className="objective"
        >
          objective 2
        </p>
        <p
          className="objective"
        >
          objective 3
        </p>
      </div>
      <div
        className="button-container"
      >
        <button
          className="pdf-button bx--btn bx--btn--ghost"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          Download as PDF
        </button>
        <button
          className="button bx--btn bx--btn--secondary"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Open tutorial
        </button>
      </div>
    </div>
    <br />
  </div>,
  <div
    aria-hidden={true}
    className="bx--tab-content"
    hidden={true}
    role="tabpanel"
    selected={false}
  >
    <div
      className="tab-container"
    >
      <div
        className="tutorial-title"
      >
        something really interesting
      </div>
      <div
        className="time-container"
      >
        <svg
          aria-hidden={true}
          className="icon"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="9"
            width="2"
            x="15"
            y="11"
          />
          <rect
            height="2"
            width="6"
            x="13"
            y="2"
          />
          <path
            d="M28,9,26.58,7.59,24.33,9.84a10.94,10.94,0,1,0,1.18,1.65ZM16,26a9,9,0,1,1,9-9A9,9,0,0,1,16,26Z"
          />
        </svg>
        <span
          className="time-text"
        >
          10 minutes
        </span>
      </div>
      <div
        className="tutorial-objectives"
      >
        <p
          className="objective"
        >
          objective 1
        </p>
        <p
          className="objective"
        >
          objective 2
        </p>
        <p
          className="objective"
        >
          objective 3
        </p>
      </div>
      <div
        className="button-container"
      >
        <button
          className="pdf-button bx--btn bx--btn--ghost"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          Download as PDF
        </button>
        <button
          className="button bx--btn bx--btn--secondary"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Open tutorial
        </button>
      </div>
    </div>
    <br />
  </div>,
]
`;

exports[`TutorialTabs component should render the expected snapshot when no tutorial data is received 1`] = `
<div
  className="tutorial-tabs-background"
/>
`;
